pandas/tests/series/test_timeseries.py

```diff
@@ -415,6 +415,24 @@ def test_pct_change(self):
             rs, (filled / filled.shift(freq="5D") - 1).reindex_like(filled)
         )
 
+    @pytest.mark.parametrize("n_1", ["1", "15"])
```

Comment:
this doesn't test the original example and I am not sure this is actually right, can you just do a simple example (which can parameterize after) that has obvious results.

Reply:
I changed the size parameters into two categories, a singular one and a plural one and
also changed the offset parameters into two categories, an anchored one and not anchored one.

And I found that the error raised when the offset(the suffix characters of the freq) in the series isn't anchored and the offset in the .pct_change is anchored.

```python3
from itertools import product

import pandas as pd

for (n_1, offset_1), (n_2, offset_2) in product(product(['1', '15'], ['D', 'BM']), product(['1', '15'], ['D', 'BM'])):
    try:
        s = pd.Series([1, 2, 3, 4, 5]*3, pd.date_range('2019-09', periods=15, freq=n_1 + offset_1))
        s.pct_change(freq=n_2 + offset_2)
    except ValueError as e:
        print((n_1, offset_1), (n_2, offset_2), '->', e)
    else:
        print((n_1, offset_1), (n_2, offset_2), '-> OK')
```

result:
```
('1', 'D') ('1', 'D') -> OK
('1', 'D') ('1', 'BM') -> cannot reindex from a duplicate axis
('1', 'D') ('15', 'D') -> OK
('1', 'D') ('15', 'BM') -> cannot reindex from a duplicate axis
('1', 'BM') ('1', 'D') -> OK
('1', 'BM') ('1', 'BM') -> OK
('1', 'BM') ('15', 'D') -> OK
('1', 'BM') ('15', 'BM') -> OK
('15', 'D') ('1', 'D') -> OK
('15', 'D') ('1', 'BM') -> cannot reindex from a duplicate axis
('15', 'D') ('15', 'D') -> OK
('15', 'D') ('15', 'BM') -> cannot reindex from a duplicate axis
('15', 'BM') ('1', 'D') -> OK
('15', 'BM') ('1', 'BM') -> OK
('15', 'BM') ('15', 'D') -> OK
('15', 'BM') ('15', 'BM') -> OK
```

Reply:
and below is the result in the branch fix-GH28664

```
('1', 'D') ('1', 'D') -> OK
('1', 'D') ('1', 'B') -> OK
('1', 'D') ('1', 'W') -> OK
('1', 'D') ('1', 'BM') -> OK
('1', 'D') ('1', 'MS') -> OK
('1', 'D') ('1', 'Q') -> OK
('1', 'D') ('15', 'D') -> OK
('1', 'D') ('15', 'B') -> OK
('1', 'D') ('15', 'W') -> OK
('1', 'D') ('15', 'BM') -> OK
('1', 'D') ('15', 'MS') -> OK
('1', 'D') ('15', 'Q') -> OK
('1', 'B') ('1', 'D') -> OK
('1', 'B') ('1', 'B') -> OK
('1', 'B') ('1', 'W') -> OK
('1', 'B') ('1', 'BM') -> OK
('1', 'B') ('1', 'MS') -> OK
('1', 'B') ('1', 'Q') -> OK
('1', 'B') ('15', 'D') -> OK
('1', 'B') ('15', 'B') -> OK
('1', 'B') ('15', 'W') -> OK
('1', 'B') ('15', 'BM') -> OK
('1', 'B') ('15', 'MS') -> OK
('1', 'B') ('15', 'Q') -> OK
('1', 'W') ('1', 'D') -> OK
('1', 'W') ('1', 'B') -> OK
('1', 'W') ('1', 'W') -> OK
('1', 'W') ('1', 'BM') -> OK
('1', 'W') ('1', 'MS') -> OK
('1', 'W') ('1', 'Q') -> OK
('1', 'W') ('15', 'D') -> OK
('1', 'W') ('15', 'B') -> OK
('1', 'W') ('15', 'W') -> OK
('1', 'W') ('15', 'BM') -> OK
('1', 'W') ('15', 'MS') -> OK
('1', 'W') ('15', 'Q') -> OK
('1', 'BM') ('1', 'D') -> OK
('1', 'BM') ('1', 'B') -> OK
('1', 'BM') ('1', 'W') -> OK
('1', 'BM') ('1', 'BM') -> OK
('1', 'BM') ('1', 'MS') -> OK
('1', 'BM') ('1', 'Q') -> OK
('1', 'BM') ('15', 'D') -> OK
('1', 'BM') ('15', 'B') -> OK
('1', 'BM') ('15', 'W') -> OK
('1', 'BM') ('15', 'BM') -> OK
('1', 'BM') ('15', 'MS') -> OK
('1', 'BM') ('15', 'Q') -> OK
('1', 'MS') ('1', 'D') -> OK
('1', 'MS') ('1', 'B') -> OK
('1', 'MS') ('1', 'W') -> OK
('1', 'MS') ('1', 'BM') -> OK
('1', 'MS') ('1', 'MS') -> OK
('1', 'MS') ('1', 'Q') -> OK
('1', 'MS') ('15', 'D') -> OK
('1', 'MS') ('15', 'B') -> OK
('1', 'MS') ('15', 'W') -> OK
('1', 'MS') ('15', 'BM') -> OK
('1', 'MS') ('15', 'MS') -> OK
('1', 'MS') ('15', 'Q') -> OK
('1', 'Q') ('1', 'D') -> OK
('1', 'Q') ('1', 'B') -> OK
('1', 'Q') ('1', 'W') -> OK
('1', 'Q') ('1', 'BM') -> OK
('1', 'Q') ('1', 'MS') -> OK
('1', 'Q') ('1', 'Q') -> OK
('1', 'Q') ('15', 'D') -> OK
('1', 'Q') ('15', 'B') -> OK
('1', 'Q') ('15', 'W') -> OK
('1', 'Q') ('15', 'BM') -> OK
('1', 'Q') ('15', 'MS') -> OK
('1', 'Q') ('15', 'Q') -> OK
('15', 'D') ('1', 'D') -> OK
('15', 'D') ('1', 'B') -> OK
('15', 'D') ('1', 'W') -> OK
('15', 'D') ('1', 'BM') -> OK
('15', 'D') ('1', 'MS') -> OK
('15', 'D') ('1', 'Q') -> OK
('15', 'D') ('15', 'D') -> OK
('15', 'D') ('15', 'B') -> OK
('15', 'D') ('15', 'W') -> OK
('15', 'D') ('15', 'BM') -> OK
('15', 'D') ('15', 'MS') -> OK
('15', 'D') ('15', 'Q') -> OK
('15', 'B') ('1', 'D') -> OK
('15', 'B') ('1', 'B') -> OK
('15', 'B') ('1', 'W') -> OK
('15', 'B') ('1', 'BM') -> OK
('15', 'B') ('1', 'MS') -> OK
('15', 'B') ('1', 'Q') -> OK
('15', 'B') ('15', 'D') -> OK
('15', 'B') ('15', 'B') -> OK
('15', 'B') ('15', 'W') -> OK
('15', 'B') ('15', 'BM') -> OK
('15', 'B') ('15', 'MS') -> OK
('15', 'B') ('15', 'Q') -> OK
('15', 'W') ('1', 'D') -> OK
('15', 'W') ('1', 'B') -> OK
('15', 'W') ('1', 'W') -> OK
('15', 'W') ('1', 'BM') -> OK
('15', 'W') ('1', 'MS') -> OK
('15', 'W') ('1', 'Q') -> OK
('15', 'W') ('15', 'D') -> OK
('15', 'W') ('15', 'B') -> OK
('15', 'W') ('15', 'W') -> OK
('15', 'W') ('15', 'BM') -> OK
('15', 'W') ('15', 'MS') -> OK
('15', 'W') ('15', 'Q') -> OK
('15', 'BM') ('1', 'D') -> OK
('15', 'BM') ('1', 'B') -> OK
('15', 'BM') ('1', 'W') -> OK
('15', 'BM') ('1', 'BM') -> OK
('15', 'BM') ('1', 'MS') -> OK
('15', 'BM') ('1', 'Q') -> OK
('15', 'BM') ('15', 'D') -> OK
('15', 'BM') ('15', 'B') -> OK
('15', 'BM') ('15', 'W') -> OK
('15', 'BM') ('15', 'BM') -> OK
('15', 'BM') ('15', 'MS') -> OK
('15', 'BM') ('15', 'Q') -> OK
('15', 'MS') ('1', 'D') -> OK
('15', 'MS') ('1', 'B') -> OK
('15', 'MS') ('1', 'W') -> OK
('15', 'MS') ('1', 'BM') -> OK
('15', 'MS') ('1', 'MS') -> OK
('15', 'MS') ('1', 'Q') -> OK
('15', 'MS') ('15', 'D') -> OK
('15', 'MS') ('15', 'B') -> OK
('15', 'MS') ('15', 'W') -> OK
('15', 'MS') ('15', 'BM') -> OK
('15', 'MS') ('15', 'MS') -> OK
('15', 'MS') ('15', 'Q') -> OK
('15', 'Q') ('1', 'D') -> OK
('15', 'Q') ('1', 'B') -> OK
('15', 'Q') ('1', 'W') -> OK
('15', 'Q') ('1', 'BM') -> OK
('15', 'Q') ('1', 'MS') -> OK
('15', 'Q') ('1', 'Q') -> OK
('15', 'Q') ('15', 'D') -> OK
('15', 'Q') ('15', 'B') -> OK
('15', 'Q') ('15', 'W') -> OK
('15', 'Q') ('15', 'BM') -> OK
('15', 'Q') ('15', 'MS') -> OK
('15', 'Q') ('15', 'Q') -> OK
```

and it also passes below test

https://github.com/pandas-dev/pandas/blob/0436570f05c3b6e7bbb7c7d8fc8fa2f28a0420a8/pandas/tests/series/test_timeseries.py#L394-L403

Reply:
I'm also lost as to what this is testing now as it doesn't match what was there originally. Can you as suggested by @jreback simplify this without parametrization and we can go from there?

Reply:
@WillAyd the original issue was caused when freq of `Series` is `D` and freq of pct_change is `BM`.
So I thought the problem is that freq of pct_change is anchored(`BM`). However, already you might know it, `1BM` is anchored but `2BM` isn't. So I added size(`n`) that can be more than 1 to test these things(`2BM`, `15W`, ...). And as you can see it from above, the anchored offset(the character part of freq) of pct_change is sufficient condition for rasing error.

And this is a non-parametrized version test. Should I replace it with this?

```python
def test_pct_change_with_duplicate_axis(self):
        s = Series([1, 2, 3, 4, 5] * 2, date_range("2019-09", periods=10, freq='D'))

        result = s.pct_change(freq='B')
        expected = s / s.shift(freq='B') - 1
        expected = expected[expected.index.isin(s.index) & ~expected.index.duplicated()]

        assert_series_equal(result, expected)
```

However in above, I'm not sure which behavior would be expected.

1. my approach(that passes existing tests but sparse though):
```
2019-09-01         NaN
2019-09-02    1.000000
2019-09-03    0.500000
2019-09-04    0.333333
2019-09-05    0.250000
2019-09-06   -0.800000
2019-09-07         NaN
2019-09-08         NaN
2019-09-09    3.000000
2019-09-10    0.250000
Freq: D, dtype: float64
```

2. .asfreq apporoach(like `s.asfreq('B').pct_change()`, but can't pass existing test `test_pct_change_periods_freq` in pandas/pandas/tests/series/test_timeseries.py):
```
2019-09-02         NaN
2019-09-03    0.500000
2019-09-04    0.333333
2019-09-05    0.250000
2019-09-06   -0.800000
2019-09-09    3.000000
2019-09-10    0.250000
Freq: B, dtype: float64
```

I need some advice about which result is expected.

Reply:
@WillAyd Hello, what should I do next?